frontend/src/components/AlterSchemaPrepForm/SchemaEditorModal.vue

```diff
@@ -212,16 +216,34 @@ const fetchDDLStatementWithUIEditor = async () => {
     }
   }
 
-  const statmentList = [];
+  const databaseEditMap: Map<DatabaseId, string> = new Map();
   if (databaseEditList.length > 0) {
-    const databaseIdList: DatabaseId[] = [];
     for (const databaseEdit of databaseEditList) {
-      databaseIdList.push(databaseEdit.databaseId);
       const statement = await editorStore.postDatabaseEdit(databaseEdit);
-      statmentList.push(statement);
+      databaseEditMap.set(databaseEdit.databaseId, statement);
     }
   }
-  return statmentList.join("\n");
+  return databaseEditMap;
+};
+
+const unsavedDialogWarning = (): Promise<string> => {
```

Comment:
Better to define a string union type or an enum type rather than string here.

Reply:
It's a temporary warning dialog. How about `Promise<string>` -> `Promise<"Close" | "NegativeClick" | "PositiveClick">`?